helper-cli/src/main/kotlin/common/Utils.kt

```diff
@@ -40,6 +42,50 @@ import java.io.File
  */
 internal typealias RepositoryPathExcludes = Map<String, List<PathExclude>>
 
+/**
+ * Return all files underneath the given [directory].
+ */
+internal fun findFilesRecursive(directory: File): List<String> {
+    require(directory.isDirectory)
+
+    val result = mutableListOf<String>()
+
+    directory.walk().forEach { file ->
+        if (!file.isDirectory) {
+            result.add(file.relativeTo(directory).path)
+        }
+    }
+
+    return result
+}
+
+/**
+ * Search the given [directory] for repositories and return a mapping from repository URLs to the relative paths where
+ * each respective repository was found.
+ */
+internal fun findRepositoryPaths(directory: File): Map<String, Set<String>> {
+    require(directory.isDirectory)
+
+    val analyzer = Analyzer(AnalyzerConfiguration(true, true))
+    val ortResult = analyzer.analyze(
+        absoluteProjectPath = directory,
+        packageManagers = emptyList(),
+        packageCurationsFile = null,
+        repositoryConfigurationFile = null
+    )
```

Comment:
You could extract lines 137-143 of `Analyzer.kt` to a function and call this instead if you only need the list of repositories, this would be much faster than actually running the analyzer:

```kotlin
        val workingTree = VersionControlSystem.forDirectory(absoluteProjectPath)
        val vcs = workingTree?.getInfo() ?: VcsInfo.EMPTY
        val nestedVcs = workingTree?.getNested()?.filter { (path, _) ->
            // Only include nested VCS if they are part of the analyzed directory.
            workingTree.getRootPath().resolve(path).startsWith(absoluteProjectPath)
        }.orEmpty()
        val repository = Repository(vcs = vcs, nestedRepositories = nestedVcs, config = repositoryConfiguration)
```

Reply:
Good point. I'd prefer doing this in a separate PR though.

Reply:
Ok, please do.